### ISSUE TITLE
The efficiency of view query is not as expected

### ISSUE THREAD

**Issue by IVitamin-C** [ClickHouse/ClickHouse#19280]
version:20.8.8.6 lts
data rows:10 billion per partition

Compared with the original table query, the query time of view query is several times slower when the query conditions are consistent.
Thanks!
The following is the operation process:

create table test.test_a1  ON CLUSTER testcluster
(
datatime  DateTime('Asia/Shanghai')
st Int64 MATERIALIZED toYYYYMMDD(datatime)
flag String 
uid String
value Int64
)
ReplicatedMergeTree(
  '/clickhouse/tables/{shard}/test.test_a1', 
  '{replica}'
)
partition toYYYYMMDD(datatime)
order by (
st*-1
,flag
,uid
,value
)
--------------------------------------------
CREATE TABLE IF NOT EXISTS test.test_a1_dis
ON CLUSTER testcluster
AS test.test_a1 
ENGINE = Distributed(testcluster,test,test_a1,rand());
---------------------------------------
CREATE VIEW IF NOT EXISTS test.test_a1_view ON CLUSTER testcluster
as 
select datatime
      ,st
      ,flag
      ,uid
      ,value
from  test.test_a1_dis
where flag='001'
-----------------------------------------
(1) Query view
select sum(1) from test.test_a1_view where uid<>''   

sum(1) :10411310174

1 rows in set .Elapsed: 4.083s. Processed 10.58 billion rows,190.29GB(2.59 billion rows/s.,46.59 GB/s.)
(2) Query Distributed
select sum(1) from test.test_a1_dis where flag='001' and uid<>'' 

sum(1) :10411310174

1 rows in set .Elapsed: 1.079s. Processed 10.58 billion rows,190.29GB(9.80 billion rows/s.,176.29 GB/s.) 

The same data query results, the efficiency gap is large.

**Comment by den-crane** [comment]
yes, because optimize_move_to_prewhere does not work against view

related https://github.com/ClickHouse/ClickHouse/issues/18561

**Comment by IVitamin-C** [comment]
So this is inconsistent with the actual situation described in the document. Is there any plan to optimize this

**Comment by den-crane** [comment]
>So this is inconsistent with the actual situation described in the document.

What document?

>Is there any plan to optimize this

Yes, someday.

**Comment by IVitamin-C** [comment]
![image](https://user-images.githubusercontent.com/35019926/105054963-89a3fd80-5aad-11eb-9dce-09a78202b285.png)

**Comment by den-crane** [comment]
that sentence is about result, not about performance.

**Comment by rienath** [comment]
This issue is older, but the other one is more active
